test-magnolia/shared/src/main/scala/zio/test/magnolia/DeriveGen.scala

```diff
@@ -66,17 +67,22 @@ object DeriveGen {
       val derive: Gen[Random with Sized, A] = gen
     }
 
-  implicit val genBoolean: DeriveGen[Boolean] = instance(Gen.boolean)
-  implicit val genByte: DeriveGen[Byte]       = instance(Gen.anyByte)
-  implicit val genChar: DeriveGen[Char]       = instance(Gen.anyChar)
-  implicit val genDouble: DeriveGen[Double]   = instance(Gen.anyDouble)
-  implicit val genFloat: DeriveGen[Float]     = instance(Gen.anyFloat)
-  implicit val genInt: DeriveGen[Int]         = instance(Gen.anyInt)
-  implicit val genLong: DeriveGen[Long]       = instance(Gen.anyLong)
-  implicit val genShort: DeriveGen[Short]     = instance(Gen.anyShort)
-  implicit val genString: DeriveGen[String]   = instance(Gen.anyString)
-  implicit val genUnit: DeriveGen[Unit]       = instance(Gen.unit)
-  implicit val genUUID: DeriveGen[UUID]       = instance(Gen.anyUUID)
+  implicit val genBoolean: DeriveGen[Boolean]             = instance(Gen.boolean)
+  implicit val genByte: DeriveGen[Byte]                   = instance(Gen.anyByte)
+  implicit val genChar: DeriveGen[Char]                   = instance(Gen.anyChar)
+  implicit val genDouble: DeriveGen[Double]               = instance(Gen.anyDouble)
+  implicit val genFloat: DeriveGen[Float]                 = instance(Gen.anyFloat)
+  implicit val genInt: DeriveGen[Int]                     = instance(Gen.anyInt)
+  implicit val genLong: DeriveGen[Long]                   = instance(Gen.anyLong)
+  implicit val genShort: DeriveGen[Short]                 = instance(Gen.anyShort)
+  implicit val genString: DeriveGen[String]               = instance(Gen.anyString)
+  implicit val genUnit: DeriveGen[Unit]                   = instance(Gen.unit)
+  implicit val genUUID: DeriveGen[UUID]                   = instance(Gen.anyUUID)
+  implicit val genLocalDateTime: DeriveGen[LocalDateTime] = instance(Gen.anyLocalDateTime)
+  implicit val genLocalDate: DeriveGen[LocalDate]         = instance(Gen.anyLocalDateTime.map(_.toLocalDate()))
+  implicit val genBigDecimal: DeriveGen[BigDecimal] = instance(
+    Gen.bigDecimal(BigDecimal(Double.MinValue), BigDecimal(Double.MaxValue))
```

Comment:
There is an issue here that `BigDecimal` is supposed to represent arbitrary precision floating point values that could be smaller than `Double.MinValue` or larger than `Double.MaxValue`. I'm not quite sure what the right answer is here since the domain is theoretically unlimited but it seems like we should be generating broader set of values than `Double` values.

Reply:
I was trying to think of what the best approach would be but couldn't really find anything that stood out.
Any suggestions?

Reply:
Why don't we do 128 bit precision (twice that of `Double`) values?

Reply:
@adamgfraser Have changed it. Please let me know if it is correct